### PR TITLE
refactor: save tx weight constants in memory and throw error when not set

### PR DESCRIPTION
**Pull request by obiyankenobi**

### CODE REVIEW

File: src/transaction.js [hunk 1]
```diff
@@ -363,6 +363,8 @@ const transaction = {
    *  'timestamp': 1,
    * }
    *
+   * @throws {ConstantNotSet} If the weight constants are not set yet
```
Comment (obiyankenobi) [inline src/transaction.js]: This function does not throw `ConstantNotSet` but it uses one that does and we don't catch the exception. 

Should we have this docstring here? @jansegre your input is also appreciated.

Comment (pedroferreira1) [reply src/transaction.js]: I think in this case where you are not catching the exception you should add in the docstring.

File: src/transaction.js [hunk 2]
```diff
@@ -600,13 +604,23 @@ const transaction = {
    *
    * @return {Object} Object with the parameters {'txMinWeight', 'txWeightCoefficient', 'txMinWeightK'}
    *
+   * @throws {ConstantNotSet} If the weight constants are not set yet
+   *
    * @memberof Transaction
    * @inner
    */
   getTransactionWeightConstants() {
-    return {'txMinWeight': parseFloat(storage.getItem('wallet:txMinWeight')),
```
Comment (msbrogli) [inline src/transaction.js]: Why don't we use a single key to all of them?
The basic usage would be:
```
const txParams = getTransactionWeightContants();
txParams.txMinWeight
```

File: src/errors.js [hunk 3]
```diff
@@ -26,3 +26,11 @@ export class AddressError extends Error {}
...
  */
 export class OutputValueError extends Error {}
+
+/**
+ * Error thrown when a constant that we get from the server is not set
+ *
+ * @memberof Errors
+ * @inner
+ */
+export class ConstantNotSet extends Error {}
```
[the remark this answers is not in the document]
Comment (obiyankenobi) [reply src/errors.js]: https://github.com/HathorNetwork/hathor-wallet-lib/issues/22

File: src/transaction.js (hunk 2 above)
Comment (obiyankenobi) [reply src/transaction.js]: Done

**Pull request merged**